packages/flutter/lib/src/cupertino/checkbox.dart

```diff
@@ -155,6 +231,23 @@ class CupertinoCheckbox extends StatefulWidget {
 
   /// The color and width of the checkbox's border.
   ///
+  /// This property can be a [WidgetStateBorderSide] that can
+  /// specify different border color and widths depending on the
+  /// checkbox's state.
+  ///
+  /// Resolves in the following states:
+  ///  * [WidgetState.pressed].
+  ///  * [WidgetState.selected].
+  ///  * [WidgetState.hovered].
+  ///  * [WidgetState.focused].
+  ///  * [WidgetState.disabled].
+  ///  * [WidgetState.error].
+  ///
+  /// If this property is not a [WidgetStateBorderSide] and it is
+  /// non-null, then it is only rendered when the checkbox's value is
+  /// false. The difference in interpretation is for backwards
+  /// compatibility.
+  ///
   /// If this property is null, then the side defaults to a one pixel wide
   /// black, solid border.
```

Comment:
Is this "one pixel wide black, solid border" also only rendered when the checkbox's value is false?

Reply:
Also, why is the default value a "one pixel wide black, solid border"? A default widget should look like the native widget, but I don't see any black solid border in the native widget, even when unselected.

Reply:
Ah... thanks for catching this. This was the current documentation before this PR that should be updated in this PR.